scheduler/task/check_k8s_cluster_status_task.go

```diff
@@ -0,0 +1,48 @@
+package task
+
+import (
+	"context"
+	"log"
+
+	"github.com/play-with-docker/play-with-docker/event"
+	"github.com/play-with-docker/play-with-docker/k8s"
+	"github.com/play-with-docker/play-with-docker/pwd/types"
+)
+
+type checkK8sClusterStatusTask struct {
+	event   event.EventApi
+	factory k8s.FactoryApi
+}
+
+func NewCheckK8sClusterStatus(e event.EventApi, f k8s.FactoryApi) *checkK8sClusterStatusTask {
+	return &checkK8sClusterStatusTask{event: e, factory: f}
+}
+
+func (c *checkK8sClusterStatusTask) Name() string {
+	return "CheckK8sClusterStatus"
+}
+
+func (c checkK8sClusterStatusTask) Run(ctx context.Context, i *types.Instance) error {
+	status := ClusterStatus{Instance: i.Name}
+
+	kc, err := c.factory.GetKubeletForInstance(i)
+	if err != nil {
+		log.Println(err)
+		c.event.Emit(CheckSwarmStatusEvent, i.SessionId, status)
+		return err
+	}
+
+	if isManager, err := kc.IsManager(); err != nil {
+		c.event.Emit(CheckSwarmStatusEvent, i.SessionId, status)
+		return err
+	} else if !isManager {
+		// Not a manager node, nothing to do for this task
+		status.IsWorker = true
+	} else {
+		status.IsManager = true
+	}
+
+	c.event.Emit(CheckSwarmStatusEvent, i.SessionId, status)
```

Comment:
I'm not sure if we should use the same event for docker swarm and kubernetes. Maybe we should use different events and the receiver decides if to do the same thing or different things?